azconfig/client/src/main/java/com/azure/azconfig/AzConfigCredentialsPolicy.java

```diff
@@ -23,11 +23,12 @@
 import java.time.ZoneOffset;
 import java.time.format.DateTimeFormatter;
 import java.util.Base64;
+import java.util.Objects;
 
 /**
  * Creates a policy that authenticates request with AzConfig service.
  */
-public final class AzConfigCredentialsPolicy extends AbstractPipelinePolicy {
+public final class AzConfigCredentialsPolicy extends HostPolicy {
```

Comment:
Why would you want to extend from HostPolicy? While it will work in this case, it seems conceptually not exactly correct to me.

Reply:
Hmm. it seemed like a policy closest in functionality to the policy that was removed. Would implementing HttpPipelinePolicy be better?

Reply:
I would do so, it's not so much code duplication. The name HostPolicy looks misleading here.

Reply:
Gotcha! I've pushed up a fix. How does it look now?